src/sonic-yang-models/yang-models/sonic-vlan-sub-interface.yang

```diff
@@ -0,0 +1,80 @@
+module sonic-vlan-sub-interface {
+
+    yang-version 1.1;
+
+    namespace  "http://github.com/Azure/sonic-vlan-sub-interface";
+
+    prefix vlan-sub-interface;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "VLAN sub interface for SONiC OS";
+
+    revision 2021-11-11 {
+        description "Initial version";
+    }
+
+    container sonic-vlan-sub-interface {
+        container VLAN_SUB_INTERFACE {
+
+            description "VLAN SUB INTERFACE part of config_db.json";
+
+            list VLAN_SUB_INTERFACE_LIST {
+
+                description "VLAN_SUB_INTERFACE part of config_db.json with vrf";
+
+                key "name";
+
+                leaf name {
+                    must "substring-before(current(), '.') = /port:sonic-port/port:PORT/port:PORT_LIST/port:name"
```

Comment:
Sub interface can be Ethernet based or PortChannel based.. This must condition which matches parent interface on PORT table will fail for Po-based subinterface. Please check.

Reply:
For the port channel, it is not included in this PR, will have it supported in the future.
For the short-name format, I have tried to add vlan sub interfaces with short name format per doc: https://github.com/Azure/SONiC/blob/master/doc/subport/sonic-sub-port-intf-hld.md, but failed, could you please elaborate more about this?

Reply:
I forgot to mention that there was a discussion before that this PR only address vlan sub interface for Ethernet based ports. @neethajohn, could you please help to clarify here?